skore/src/skore/sklearn/_estimator/report.py

```diff
@@ -118,6 +121,19 @@ def _initialize_state(self):
         self._cache = {}
         self._ml_task = _find_ml_task(self._y_test, estimator=self._estimator)
 
+    def _copy_estimator(self, estimator):
+        try:
+            return copy.deepcopy(estimator)
+        except Exception as e:
+            warnings.warn(
+                "Deepcopy failed; using estimator as-is. "
+                "Be aware that modifying the estimator outside of "
+                f"{self.__class__.__name__} will modify the internal estimator. "
+                "Consider using a FrozenEstimator from scikit-learn to prevent this. "
+                f"Original error: {e}",
+                stacklevel=1,
+            )
```

Comment:
```suggestion
            )
            return estimator
```

Reply:
It would look that it is easier to understand what we intend to even it is no-op at the end.

Reply:
done